pulp_smash/tests/rpm/api_v2/test_noop.py

```diff
@@ -0,0 +1,264 @@
+        7. publish with overrides_config (identical to step 5)
+        8. assert publish is noop
+
+        """
+        super(NOOPTestCase1, cls).setUpClass()
+        cls.responses = {}
+        cls.tasks = {}
+        cls.xmldata = {}
+        if selectors.bug_is_untestable(1724, cls.cfg.version):
+            raise unittest2.SkipTest('https://pulp.plan.io/issues/1724')
+
+        # Create and sync a repository.
+        client = api.Client(cls.cfg)
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_FEED_URL
+        repo_href = client.post(REPOSITORY_PATH, body).json()['_href']
+        cls.resources.add(repo_href)  # mark for deletion
+        cls.responses['sync'] = utils.sync_repo(cls.cfg, repo_href)
+
+        # Add a distributor and publish it.
+        distributor = client.post(
+            urljoin(repo_href, 'distributors/'),
+            gen_distributor(),
+        ).json()
+        cls.responses['first publish'] = client.post(
+            urljoin(repo_href, 'actions/publish/'),
+            {'id': distributor['id']},
+        ).json()
+        next(api.poll_spawned_tasks(cls.cfg, cls.responses['first publish']))
+        cls.xmldata['publish1'] = get_repo_md(client, distributor)
+
+        # no op publish
+        cls.responses['second publish'] = client.post(
+            urljoin(repo_href, 'actions/publish/'),
+            {'id': distributor['id']},
+        )
+        next(api.poll_spawned_tasks(cls.cfg, cls.responses['second publish']))
+        cls.xmldata['publish2'] = get_repo_md(client, distributor)
+
+        # publish with overrides, that shouldn't be noop
+        cls.responses['third publish'] = client.post(
+            urljoin(repo_href, 'actions/publish/'),
+            {'id': distributor['id'],
+             'config_overrides': {'relative_url': '/foo/bar/repo'}},
+        )
+        next(api.poll_spawned_tasks(cls.cfg, cls.responses['third publish']))
+        cls.xmldata['publish3'] = get_repo_md(client, distributor)
+
+        # publish with overrides, that shouldn't be noop
+        cls.responses['fourth publish'] = client.post(
+            urljoin(repo_href, 'actions/publish/'),
+            {'id': distributor['id'],
+             'config_overrides': {'relative_url': '/foo/bar/repo'}},
+        )
+        next(api.poll_spawned_tasks(cls.cfg, cls.responses['forth publish']))
+        cls.xmldata['publish4'] = get_repo_md(client, distributor)
+
+        # publish with overrides (identical as in publish3),
+        # should be noop
+        for publish in ('first publish', 'second publish', 'third publish',
+                        'fourth publish'):
+            report = cls.responses[publish]
+            cls.tasks[publish] = next(api.poll_spawned_tasks(cls.cfg,
+                                                             report))
+
+    def test_01_op_publish(self):
+        """Test if first publish is operational."""
+        self.assertTrue(isinstance(
```

Comment:
When this test fails, it'll produce a message like "False is not True." Using an assertion like `assertIsInstance` will produce error messages like "str is not dict."

You don't need to fix this right now. But it's the sort of thing I'll change in a follow-up commit.